File: ListaCompra/Python/Lista_Compra.py
# Reto: Lista de la Compra
#
# Crea un programa que simule una lista de la compra. El programa debe permitir al usuario agregar elementos a la lista, eliminar elementos y 
# mostrar la lista completa. Además, debería verificar si un artículo ya está en la lista antes de agregarlo nuevamente.
# Puedes implementar esto en cualquier lenguaje de programación que prefieras. Siéntete libre de agregar funciones adicionales según lo consideres necesario.
# Variables
lista_compra = []
salir = True

while salir:
    print("|-----------------------|")
    print("|1. Introducir elemento.|")
    print("|2. Borrar elemento.    |")
    print("|3. Mostrar listado.    |")
    print("|4. Salir del programa. |")
    print("|-----------------------|")
    opcion = int(input("Introduce una opción de la consola:"))

    if opcion == 1:
        # Introducimos el elemento
        print("Has seleccionado la opcion de introducir un elemento")
        intro_elem = input("Introduce el elemento: ")
        if intro_elem not in lista_compra:
            lista_compra.append(intro_elem)
            print(f"{intro_elem} agregado a la lista.")
        else:
            print(f"{intro_elem} ya está en la lista.")
    elif opcion == 2:
        print("Has seleccionado la opcion de eliminar un elemento")
        intro_elem = input("Introduce el elemento que deseas eliminar: ")
        if intro_elem in lista_compra:
            lista_compra.remove(intro_elem)
            print(f"{intro_elem} eliminado de la lista.")
        else:
            print(f"{intro_elem} no existe en la lista.")
    elif opcion == 3:
        print("Lista de compras:")
        for elemento in lista_compra:
            print(elemento)
    elif opcion == 4:
        print("El programa ha finalizado.")
        salir = False
        break
    else:
        print("Introduce una opcion valida")

File: ListaCompra/JAVA/ListaCompra.java
package JAVA;

import java.util.ArrayList;
import java.util.Scanner;

/**
 * ListaCompra
 */
public class ListaCompra {

    public static void main(String[] args) {

        //Encerramos el scanner en un try-catch para gestionar los posibles errores
        try(Scanner sc = new Scanner(System.in)) {
            //Variables
            int respuesta = 0;
            ArrayList<String> listaDeCompras = new ArrayList<>();
            String introElem = "";


            do {
                System.out.println("|----------------------|");
                System.out.println("|1. Insertar elemento. |");
                System.out.println("|2. Borrar elemento.   |");
                System.out.println("|3. Mostrar elementos. |");
                System.out.println("|4. Salir del programa.|");
                System.out.println("|----------------------|");
                respuesta = sc.nextInt();

                switch (respuesta) {
                    case 1:
                        //Insertar
                        //Pedimos al usuario que introduzca un elemento a la lista
                        System.out.println("Has seleccionado la opcion de introducir un elemento.");
                        System.out.println("Introduce un elemento al listado: ");
                        sc.nextLine();
                        introElem = sc.nextLine();//Capturamos el elemento
                        //Lo insertamos en el array
                        listaDeCompras.add(introElem);
                        System.out.println("El elemento ha sido introducido con éxito");//Mensaje de que ha tenido exito
                    break;
                    case 2:
                        //Eliminar
                        //Primero le mostramos la lista para que vea que puede borrar
                        for(String elemento:listaDeCompras){
                            System.out.println(elemento);
                        }
                        //Ahora le preguntamos que desea borrar
                        System.out.println("Escribe el elemento que deseas borrar de la lista:");
                        sc.nextLine();
                        introElem= sc.nextLine();
                        listaDeCompras.remove(introElem); //Eliminamos
                        System.out.println("El elemento ha sido eliminado con exito.");
                    break;
                    case 3:
                        //Mostrar elementos
                        for(String elemento:listaDeCompras){
                            System.out.println(elemento);
                        }
                    break;
                    case 4:
                        //Salir
                        System.out.println("El programa ha finalizado.");
                    break;
                    default:
                        System.out.println("El numero introducido esta fuera del rango permitido, por favor introduce un numero del 1-4.");
                        System.out.println("El programa ha finalizado.");
                    break;
                }

            } while ((respuesta > 0) && (respuesta < 5));

        } catch (Exception e) {
            System.out.println("Por favor, introduce solamente números del 1-4.");
        }

    }
}